pkg/api/testapi/testapi.go

```diff
@@ -79,6 +79,10 @@ func init() {
 	Extensions = Groups[extensions.GroupName]
 }
 
+func (g TestGroup) ContentConfig() (string, *unversioned.GroupVersion, runtime.Codec) {
+	return "application/json", g.GroupVersion(), g.Codec()
+}
+
```

Comment:
I wanted to keep codec small enough that it was just two methods - encode
and decode.  Mime type is definitely something that the person who returns
the codec knows - the struct returned there looks a lot like
ContentConfig.  We could potentially move a contentconfig like struct into
runtime - but then would we want to import it into our client config struct?

The other case I had today was with the streaming codec (which will be a
follow up pull after split codec lands) - it feels reasonable for Encoder
to know how return an io.Writer that does framing (for streaming protocols)
- having both an Encoder and StreamingEncoder seems incorrect.  The short
term choice I made was to have Encoder support returning a streaming
encoder as an optional interface, but this means that wrapping encoders
(something we want to encourage) becomes harder.  An alternate choice would
be to have the mechanism for getting codecs to also allow you to say "i
want the streaming codec for this codec" by passing a codec back.

On Thu, Jan 7, 2016 at 1:40 PM, krousey <notifications@github.com> wrote:

> In pkg/api/testapi/testapi.go
> <https://github.com/kubernetes/kubernetes/pull/19105#discussion_r49108395>
> :
>
> > @@ -79,6 +79,10 @@ func init() {
> >  	Extensions = Groups[extensions.GroupName]
> >  }
> >
> > +func (g TestGroup) ContentConfig() (string, *unversioned.GroupVersion, runtime.Codec) {
> > +	return "application/json", g.GroupVersion(), g.Codec()
> > +}
> > +
>
> Hmm.. To me, this use case suggest that perhaps content type should be
> part of the codec. Thoughts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/19105/files#r49108395>.
>